src/flows/pipes/RawFluxEditor/FunctionsList/DynamicFunctionsList.tsx

```diff
@@ -24,19 +24,44 @@ const sortFuncs = (a, b) => {
     return a.package.toLowerCase() < b.package.toLowerCase() ? -1 : 1
   }
 }
+const hoveredFunctions = new Set<string>()
+let recordedFunctions = new Set<string>()
 
 const DynamicFunctionsList: FC<Props> = ({onSelect}) => {
+  const [eventSearchTerm, setEventSearchTerm] = useState('')
+  const [termRecorded, setTermRecorded] = useState('')
+  const [tooltipPopup, setTooltipPopup] = useState(false)
+  const [hoverdFunction, setHoverdFunction] = useState('')
+
   const dispatch = useDispatch()
   const fluxFunctions = useSelector(getAllFluxFunctions)
+
   useEffect(() => {
     if (fluxFunctions.length === 0) {
       dispatch(getFluxPackages())
     }
   }, [])
 
+  useEffect(() => {
+    if (tooltipPopup && eventSearchTerm !== termRecorded) {
+      hoveredFunctions.clear()
+      event('flux.function.searched', {searchTerm: eventSearchTerm})
+      setTermRecorded(eventSearchTerm)
+    }
+    setTooltipPopup(false)
+    if (tooltipPopup) {
+      hoveredFunctions.add(hoverdFunction)
+
+      if (recordedFunctions.size !== hoveredFunctions.size) {
+        event('hovered.flux.function', hoveredFunctions)
+        recordedFunctions = new Set([...hoveredFunctions])
```

Comment:
i think `recordedFunctions.add(hoveredFunction)` is easier, right? why re-create the whole set again just to add a hovered function?

Reply:
yes much easier than what i have! done!!